File: Daily Test/Java/Circular Order - Skip Odd & Even 22.10.2025.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] reg = new int[n];
        for (int i = 0; i < n; i++) reg[i] = sc.nextInt();
        int q = sc.nextInt();
        int[] marks = new int[q];
        for (int i = 0; i < q; i++) marks[i] = sc.nextInt();

        int[] res = new int[n];
        List<Integer> order = new ArrayList<>();

        for (int i = 0; i < n; i++) order.add(i);
        for (int i = 0; i < n; i++) if (reg[i] % 2 == 0) order.add(i);
        for (int i = 0; i < n; i++) if (reg[i] % 2 != 0) order.add(i);

        int order_len = order.size();

        for (int i = 0; i < q; i++) {
            res[order.get(i % order_len)] += marks[i];
        }

        for (int val : res) System.out.print(val + " ");
        System.out.println();
        sc.close();
    }
}

File: Daily Test/Python/Circular Order - Skip Odd & Even 22.10.2025.py
n = int(input())
reg = list(map(int, input().split()))
q = int(input())
marks = list(map(int, input().split()))

res = [0] * n
order = []
order += range(n)
order += [i for i in range(n) if reg[i] % 2 == 0]
order += [i for i in range(n) if reg[i] % 2]

for i in range(q):
    res[order[i % len(order)]] += marks[i]

print(*res)